### PR TITLE
[4.0] Check out improvements related to nullable columns

### PR DESCRIPTION
**Pull request by SharkyKZ**
Closes https://github.com/joomla/joomla-cms/issues/30731.

### Summary of Changes

Adds nullable checks for proper handling of 0 value in `com_checkin`.

### Testing Instructions

Set `checked_out` field value to `0` in one of core tables, e.g by running this query:

    UPDATE #__menu SET checked_out=0 WHERE client_id=0

Go to `com_checkin`.

It would also be good to test that check in still works with 3rd party components that don't use null values.

### Actual result BEFORE applying this Pull Request

`#__menu` table not listed as having checked out items.

### Expected result AFTER applying this Pull Request

`#__menu` table listed as having checked out items and items can be checked in properly.

### Documentation Changes Required

No.

### CODE REVIEW

File: administrator/components/com_checkin/src/Model/CheckinModel.php [hunk 1]
```diff
@@ -107,26 +107,31 @@ public function checkin($ids = array())
...
 					->bind(':checkouttime', $nullDate);
 			}
 
+			if ($fields['checked_out']->Null === 'YES')
+			{
+				$query->where($db->quoteName('checked_out') . ' IS NOT NULL');
+			}
+			else
+			{
+				$query->where($db->quoteName('checked_out') . ' != 0');
```
Comment (richard67) [inline administrator/components/com_checkin/src/Model/CheckinModel.php]: Not `> 0`?

Comment (richard67) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: Just a question, am not sure if that would be right.

Comment (SharkyKZ) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: Changed it back. Not sure if there's a difference in this case. Maybe if someone is using signed integers?

Comment (richard67) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: To be honest, I don't even know from scratch what the values of the checked_out column means. I would have to look that up. What value shall it have when an item is not checked out on J3, and what on J4?

Comment (SharkyKZ) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: It's the user ID. In J3 default is 0, in J4 it's `null`. 3PDs *could* use some arbitrary value like `-1` for whatever reasons. So I'll just leave it as it was.

Comment (richard67) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: Ok.

Comment (mahagr) [reply administrator/components/com_checkin/src/Model/CheckinModel.php]: Latest code looks good to me. It will have the new behavior for nullable fields and use the old logic for the rest.

**Comment by richard67** [comment]
@mahagr Could you test this PR and mark the test result in the issue tracker? Thanks in advance.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 97d2867d8ef01af4b76775d577b8085763047157<br /><br />Tested successfully in Beta5-dev of 23 September.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30747">issues.joomla.org/tracker/joomla-cms/30747</a>.</sub>

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 97d2867d8ef01af4b76775d577b8085763047157<br /><br />Works as described. Not tested third party components.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30747">issues.joomla.org/tracker/joomla-cms/30747</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30747">issues.joomla.org/tracker/joomla-cms/30747</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Comment by richard67** [comment]
RTC
